dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/Child.java

```diff
@@ -36,7 +36,14 @@ public void execute(final Namespace ns, final DockerfileGitHubUtil dockerfileGit
 
         log.info("Retrieving repository and creating fork...");
         GHRepository repo = dockerfileGitHubUtil.getRepo(ns.get(Constants.GIT_REPO));
-        GHRepository fork = dockerfileGitHubUtil.closeOutdatedPullRequestAndFork(repo);
+        GHRepository fork;
+        try {
+             fork = dockerfileGitHubUtil.closeOutdatedPullRequestAndFork(repo);
+        } catch (IOException e) {
+            log.info("Could not fork {}", repo.getFullName(), e);
+            String msg = String.format("Unable to fork %s. Please make sure that the repo is forkable.");
+            throw new IllegalAccessException(msg);
```

Comment:
Yup. Added a test case.

Reply:
We actually don't know if the `IOException` was exactly the case you're solving for if I understand correctly so switching to that exception doesn't seem correct.

Reply:
Library throws IOException if any failure happens when forking. Anyways now I updated this to check for return value.

Reply:
Why are we throwing an `IllegalAccessException` if it may not be related to this use case? It could be a network error or other things as well.

Reply:
There may be a good reason, I'm just asking BTW :)

Reply:
In this case, we need to stop the execution flow and return if we are unable to create a PR. Would you recommend to throw any other exception?

Reply:
Is there a reason you couldn't simply log a message and return instead of throwing an exception? I guess I don't see a need to introduce another exception when you can handle flow control without it.

Reply:
Makes sense. Updated PR.